### PR TITLE
Product history zth

### PR DESCRIPTION
**Pull request by zachholcomb**
# PR Description
Please include a summary of the change:
Adds a joins table between shipments and products
Adds returned boolean columbn to shipmentProducts table
:100: Passing All Tests

Commit Emojis:
🥇 

If it is a new feature, describe the feature and the iteration of the project spec it addresses.
This sets up the relationship between a shipment and all the products that might be contained in a shipment. It also gives each product on that shipment a returned boolean which will default to false.

Note any questions or struggles as well.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/api/src/modules/product_shipments/mutations.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Imports
+import { GraphQLInt } from 'graphql'
```
Comment (zachholcomb) [inline code/api/src/modules/product_shipments/mutations.js:2]: I just set up these files, I will be adding a lot to them soon, but need to setup the data so I can test them

File: code/api/src/modules/product_shipments/query.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Imports
+import { GraphQLInt, GraphQLList } from 'graphql'
```
Comment (zachholcomb) [inline code/api/src/modules/product_shipments/query.js:2]: Same here

File: code/api/src/modules/product/model.js [hunk 3]
```diff
@@ -22,4 +22,10 @@ module.exports = function(sequelize, DataTypes) {
       type: DataTypes.TEXT
     }
   })
+
+  Product.associate = function(models) {
+    Product.belongsToMany(models.Shipment, { through: models.ProductShipment })
```
Comment (zachholcomb) [inline code/api/src/modules/product/model.js:27]: according to sequelize docs this is how you set up a many to many relationship, so hopefully it works when I start seeding and testing

File: code/api/src/migrations/10-shipment-returned.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+'use strict';
```
Comment (zachholcomb) [inline code/api/src/migrations/10-shipment-returned.js:1]: I separated out the migrations because I forgot about the returned column

Comment (atkinson-daniel) [reply code/api/src/migrations/10-shipment-returned.js:1]: No worries this looks good!

File: code/api/src/modules/product/model.js (hunk 3 above)
Comment (atkinson-daniel) [inline code/api/src/modules/product/model.js:27]: Nice this looks solid

**Review by atkinson-daniel — APPROVED** [review]
Looks good

File: code/api/src/modules/product_shipments/model.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    shipmentId: {
+      type: DataTypes.INTEGER
+    },
+    returned: {
+      type: DataTypes.BOOLEAN,
+      defaultValue: false
+    }
+  })
+
+  return ProductShipment
```
Comment (rachael-t) [inline code/api/src/modules/product_shipments/model.js:18]: Looks good! Thanks for adding the defaultValue of false.

File: code/api/src/modules/product_shipments/mutations.js (hunk 1 above)
Comment (rachael-t) [reply code/api/src/modules/product_shipments/mutations.js:2]: Sounds good! Thanks for clarifying that.

File: code/api/src/modules/product_shipments/query.js (hunk 2 above)
Comment (rachael-t) [reply code/api/src/modules/product_shipments/query.js:2]: 👍

**Review by rachael-t — APPROVED** [review]
Great job getting this setup!

**Pull request merged**